metrics/config.go

```diff
@@ -72,6 +78,25 @@ func getMetricsConfig(m map[string]string, domain string, component string, logg
 		mc.stackdriverProjectID = m[stackdriverProjectIDKey]
 	}
 
+	// If reporting period is specified, use the value from the configuration.
+	// If not, set a default value based on the selected backend.
+	// Each exporter makes different promises about what the lowest supported
+	// reporting period is. For Stackdriver, this value is 1 minute.
+	// For Prometheus, we will use a lower value since the exporter doesn't
+	// push anything but just responds to pull requests, and shorter durations
+	// do not really hurt the performance and we rely on the scraping configuration.
+	if repStr, ok := m[reportingPeriodKey]; ok && len(repStr) != 0 {
+		if repInt, err := strconv.Atoi(repStr); err == nil {
+			mc.reportingPeriod = time.Duration(repInt) * time.Second
+		} else {
+			return nil, fmt.Errorf("Invalid reporting-period-seconds value \"%s\"", repStr)
+		}
+	} else if mc.backendDestination == Stackdriver {
+		mc.reportingPeriod = 60 * time.Second
+	} else if mc.backendDestination == Prometheus {
```

Comment:
What happen if  `backendDestination` is neither `Stackdriver` nor `Prometheus`.

Reply:
It will return an error in the if block above (line 71).